config/management/src/waypoint.rs

```diff
@@ -41,18 +43,86 @@ impl CreateWaypoint {
 
         if let Some(remote) = self.secure_backends.remote {
             let mut remote: BoxedStorage = remote.try_into()?;
-            remote
-                .available()
-                .map_err(|e| Error::RemoteStorageUnavailable(e.to_string()))?;
-
-            let waypoint_value = Value::String(waypoint.to_string());
-            remote
-                .set(libra_global_constants::WAYPOINT, waypoint_value)
-                .map_err(|e| {
-                    Error::RemoteStorageWriteError(libra_global_constants::WAYPOINT, e.to_string())
-                })?;
+            InsertWaypoint::insert_waypoint_to_backend(
+                &waypoint,
+                &mut remote,
+                RelativePosition::Remote,
+            )?;
         }
-
         Ok(waypoint)
     }
 }
+
+#[derive(Debug, StructOpt)]
+pub struct InsertWaypoint {
+    #[structopt(flatten)]
+    secure_backends: SecureBackends,
+
+    #[structopt(long)]
+    waypoint: Option<String>,
+}
+
+impl InsertWaypoint {
+    pub fn execute(self) -> Result<Waypoint, Error> {
+        if self.waypoint.is_some() && self.secure_backends.remote.is_some() {
+            return Err(Error::CommandArgumentError(
+                "only one of --waypoint and --remote can be provided".to_string(),
+            ));
+        }
+        let mut waypoint_string = self.waypoint.clone();
+        if let Some(remote) = self.secure_backends.remote {
+            let remote_backend: BoxedStorage = remote.try_into()?;
+            waypoint_string = Some(
+                remote_backend
+                    .get(constants::WAYPOINT)
+                    .and_then(|v| v.value.string())
+                    .map_err(|e| {
+                        Error::RemoteStorageReadError(constants::WAYPOINT, e.to_string())
+                    })?,
+            );
+        }
+
+        if let Some(waypoint_string) = waypoint_string {
+            let waypoint = Waypoint::from_str(&waypoint_string)
+                .map_err(|e| Error::UnexpectedError(e.to_string()))?;
+            let mut backend: BoxedStorage = self.secure_backends.local.try_into()?;
+            InsertWaypoint::insert_waypoint_to_backend(
+                &waypoint,
+                &mut backend,
+                RelativePosition::Local,
+            )?;
+            Ok(waypoint)
+        } else {
+            Err(Error::CommandArgumentError(
+                "please provide either --waypoint or --remote".to_string(),
+            ))
+        }
```

Comment:
seems weird to have this twice, use an expect / panic if we get here... if we need this

Reply:
fixed by the tip provded!